docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-1339/README.md

```diff
@@ -0,0 +1,115 @@
+# CWE-1339: Insufficient Precision or Accuracy of a Real Number
+
+Avoid floating-point and use integers or the `decimal` module to ensure precision in applications that require high accuracy, such as in financial or banking computations.
+
+In Python, floating-point types are constrained by a fixed number of binary mantissa bits, typically allowing for around seven decimal digits of precision (24-bit values). Consequently, they are not well-suited for representing surds, such as `√7` or `π` with full accuracy. Additionally, due to their binary nature, floating-point types are incapable of exactly terminating decimals in base 10, such as 0.3, which has a repeating binary representation.
+
+To ensure precision in applications requiring high accuracy, ushc as in financial or banking computations, it is recommended to avoid using floating-point types. Instead, integers or more precise data types like the `Decimal`class should be employed.
+
+## Non-compliant Code Example
+
+This `noncompliant01.py` demonstrates the use of floating-point arithmetic to simulate purchasing items and subsequent
+
+*[noncompliant01.py](noncompliant01.py):*
+
+```py
+# SPDX-FileCopyrightText: OpenSSF project contributors
+# SPDX-License-Identifier: MIT
+""" Non-compliant Code Example """
+BALANCE = 3.00
+ITEM_COST = 0.33
+ITEM_COUNT = 5
+ 
+print(
+    f"{str(ITEM_COUNT)} items bought, ${ITEM_COST} each. "
+    f"Current account balance: "
+    f"${str(BALANCE - ITEM_COUNT * ITEM_COST)}"
+)
+```
+
+Because the value of `ITEM_COST`(0.33) cannot be precisely represented in Python due to its nature as a terminating decimal in base 10, the resulting output was as follows:

```

Comment:
missing space
```suggestion
Because the value of `ITEM_COST` (0.33) cannot be precisely represented in Python due to its nature as a terminating decimal in base 10, the resulting output was as follows:
```